grails-app/controllers/org/pih/warehouse/product/ProductGroupController.groovy

```diff
@@ -151,6 +151,9 @@ class ProductGroupController {
                 redirect(controller: "productGroup", action: "list")
             } else {
                 println productGroupInstance.errors
+                // Refresh the instance from db, to avoid returning to the view productGroupInstance
+                // that is persisted in Hibernate session with the binded properties that didn't pass the validation
+                productGroupInstance.refresh()
```

Comment:
I think this is a common thing. If something does not pass the validation then you show a validation exception without reverting to the good state, but the user has to correct it.

Reply:
The case was that by leaving it as it was, the name was also updated in the header while it should display the actual name of the product group, not the one that you tried to edit it for:
![Screenshot from 2023-03-10 10-19-45](https://user-images.githubusercontent.com/93163821/224276364-845cc347-4cde-4655-a36a-e09480704036.png)

If I wanted to edit it for `test_group` (I already have `test_group`), if the validation occurs, I would see the `test_group` in the header, not the `test_group888`